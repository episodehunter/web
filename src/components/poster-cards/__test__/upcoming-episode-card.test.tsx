import * as React from 'react'
import * as renderer from 'react-test-renderer'
import {
  UpcomingEpisodeCardComponent,
  formatEpisodeAirDate
} from '../upcoming-episode-card'

test('Print TBA when air date is null', () => {
  // Arrange
  const airDate = null

  // Act
  const result = formatEpisodeAirDate(airDate)

  // Assert
  expect(result).toBe('TBA')
})

test('Print the date when air date is not null', () => {
  // Arrange
  const airDate = new Date('1523-06-06')

  // Act
  const result = formatEpisodeAirDate(airDate)

  // Assert
  expect(result).toBe('6 JUNE')
})

it('Poster renders correctly', () => {
  const tree = renderer
    .create(
      <UpcomingEpisodeCardComponent
        episodeAirDate={new Date('1523-06-06')}
        showId={5}
        showName="Some show name"
        tvdbId={5}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
