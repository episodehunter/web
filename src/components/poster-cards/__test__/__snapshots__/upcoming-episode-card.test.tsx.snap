// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster renders correctly 1`] = `
<a
  className="sc-bxivhb goOUWs sc-bdVaJa cxUmkD"
  onClick={[Function]}
>
  <img
    defaultImage="https://www.placecage.com/g/185/273"
    height={273}
    image="https://d1lolx4ilifvdr.cloudfront.net/poster/185x273/5.jpg"
    width={185}
  />
  <div
    className="sc-ifAKCX jHqQRB"
  >
    <div
      className="sc-bwzfXH uTxCW"
    >
      Some show name
    </div>
    <div
      className="sc-htpNat buBeEl"
    >
      6 JUNE
    </div>
  </div>
</a>
`;
